cloudomate/hoster/vps/clientarea.py

```diff
@@ -1,256 +1,110 @@
 
-    def _extract_services(self, html):
-        soup = BeautifulSoup(html, 'lxml')
-        rows = soup.find('table', {'id': 'tableServicesList'}).tbody.findAll('tr')
-        self.services = []
-        for row in rows:
-            tds = row.findAll('td', recursive=False)
-            classes = tds[3].span['class']
-            status = ''
-            for c in classes:
-                if 'status-' in c:
-                    status = c.split('status-')[1]
-
-            self.services.append({
-                'id': tds[4].a['href'].split('id=')[1],
-                'product': tds[0].strong.text,
-                'price': tds[1].contents[0],
-                'term': tds[1].contents[2],
-                'next_due_date': tds[2].text[0:10],
-                'status': status,
-                'url': self.clientarea_url + tds[4].a['href'].split('.php')[1],
-            })
-        return self.services
-
-    def _get_vserverid(self, url):
-        page = self.browser.open(url)
-        match = re.search(r'vserverid = (\d+)', page.text)
-        return match.group(1)
-
-    def get_client_data_ip(self, client_data_url):
-        """
-        Get the ip of specified service through client_data.php. For some clientarea providers this is the way to obtain 
-        the IP address.
-        :param client_data_url: the URL pointing towards the clientarea's client_data.php
-        :return: the IP address
-        """
-        data = self.get_client_data(client_data_url)
-        return data['mainip']
+    def get_services_first(self):
+        return self.get_services()[0]
 
-    def get_client_data(self, client_data_url):
-        service = self.get_specified_service()
-        self._ensure_active(service)
-        vserverid = self._get_vserverid(service['url'])
-        millis = int(round(time.time() * 1000))
-        page = self.browser.open(client_data_url + '?vserverid=%s&_=%s' % (vserverid, millis))
-        return json.loads(page.get_data())
+    def _parse_service_row(self, row):
+        columns = row.findAll('td')
 
-    def get_client_data_info_dict(self, client_data_url):
-        data = self.get_client_data(client_data_url)
-        return OrderedDict([
-            ('Used bandwidth', data['bandwidthused'] + ' / ' + data['bandwidthtotal']),
-            ('Used memory', data['memoryused'] + ' / ' + data['memorytotal']),
-            ('Used storage', data['hddused'] + ' / ' + data['hddtotal']),
-            ('IP address', data['mainip']),
-        ])
+        name = columns[0].strong.text
 
-    def get_specified_service(self):
-        """
-        If number is set through config or cli, return the service according to the number, else return the first 
-        service.
-        :return: the chosen service
-        """
-        number = self._get_number()
-        self._services()
-        self._verify_number(number)
-        return self.services[number]
+        price = columns[1].text
+        i = price.index('.')
```

Comment:
Please rename these variables (`i`, `p`,  `c` etc) to something more meaningful.